File: plugins/turbo-plugin-code-comment/skills/tp-js-comment/assets/example-with-comments.ts
// 此檔案示範 js-comment skill 要求的各類型 JS/TS 文件與說明註解格式。
// 範例以正體中文撰寫，除非 repository 已建立英文習慣。

/**
 * 代表裝置訂閱方案的結構，記錄方案識別碼與允許的裝置上限。
 */
export interface SubscriptionPlan {
  /** 方案唯一識別碼，由後端生成後傳入，不可由前端修改。 */
  id: string;
  /** 此方案允許同時登入的最大裝置數量。 */
  maxDeviceCount: number;
}

/**
 * 管理使用者訂閱狀態，提供裝置新增與到期日計算功能。
 * 作為領域服務使用，不直接存取 API；由呼叫端負責資料持久化。
 */
export class SubscriptionService {
  /** 此服務所管理的訂閱方案，決定裝置上限等約束。 */
  private readonly plan: SubscriptionPlan;

  /**
   * 初始化 SubscriptionService。
   * @param plan 要套用的訂閱方案，用於決定裝置上限等規則。
   */
  constructor(plan: SubscriptionPlan) {
    this.plan = plan;
  }

  /**
   * 根據方案月數計算從今日起算的到期日。
   * @param planMonths 訂閱月數，必須為正整數。
   * @returns 到期日的 Date 物件，時間設為當天 23:59:59 UTC。
   * @throws {RangeError} 當 planMonths 小於等於零時擲出。
   */
  calculateExpiryDate(planMonths: number): Date {
    if (planMonths <= 0) throw new RangeError('planMonths 必須大於零。');

    const base = new Date();
    base.setUTCMonth(base.getUTCMonth() + planMonths);

    // 到期日設為當天最後一刻，確保使用者可使用至該日結束
    base.setUTCHours(23, 59, 59, 0);
    return base;
  }

  /**
   * 嘗試將裝置加入訂閱，若已達裝置上限則拒絕並回傳 false。
   * @param deviceIds 目前已綁定的裝置識別碼清單（直接修改此陣列）。
   * @param deviceId 要新增的裝置識別碼，不得為空字串。
   * @returns 成功新增時回傳 true；已達上限時回傳 false。
   * @throws {Error} 當 deviceId 為空白時擲出。
   */
  tryAddDevice(deviceIds: string[], deviceId: string): boolean {
    if (!deviceId.trim()) throw new Error('裝置 ID 不得為空白。');

    if (deviceIds.length >= this.plan.maxDeviceCount) return false;

    deviceIds.push(deviceId);
    return true;
  }
}

/**
 * 計算兩個訂閱方案中裝置上限較高者，並回傳對應方案。
 * 若兩者相等，優先回傳第一個方案。
 * @template T 方案型別，必須實作 SubscriptionPlan。
 * @param planA 第一個候選方案。
 * @param planB 第二個候選方案。
 * @returns 裝置上限較高的方案；若相等則回傳 planA。
 */
export function selectHigherCapacityPlan<T extends SubscriptionPlan>(planA: T, planB: T): T {
  // 相等時回傳 planA，維持呼叫端對結果的可預測性
  return planB.maxDeviceCount > planA.maxDeviceCount ? planB : planA;
}

File: plugins/turbo-plugin-code-comment/skills/tp-csharp-comment/assets/example-with-comments.cs
// 此檔案示範 csharp-comment skill 要求的各類型 C# 文件與說明註解格式。
// 範例以正體中文撰寫，除非 repository 已建立英文習慣。

using System;
using System.Collections.Generic;

namespace ExampleNamespace
{
    /// <summary>
    /// 管理使用者訂閱狀態，並提供訂閱到期日的計算與更新操作。
    /// 作為領域服務使用，不直接依賴資料庫；由呼叫端負責持久化。
    /// </summary>
    public class SubscriptionService
    {
        /// <summary>每個方案允許的最大裝置數量上限，超過時拒絕新增。</summary>
        private readonly int _maxDeviceCount;

        /// <summary>用於計算到期日的時鐘抽象，允許測試時替換為固定時間。</summary>
        private readonly ISystemClock _clock;

        /// <summary>
        /// 初始化 <see cref="SubscriptionService"/>。
        /// </summary>
        /// <param name="maxDeviceCount">此方案允許的最大裝置數量，必須大於零。</param>
        /// <param name="clock">系統時鐘介面，用於取得目前時間。</param>
        /// <exception cref="ArgumentOutOfRangeException">當 <paramref name="maxDeviceCount"/> 小於等於零時擲出。</exception>
        public SubscriptionService(int maxDeviceCount, ISystemClock clock)
        {
            if (maxDeviceCount <= 0)
                throw new ArgumentOutOfRangeException(nameof(maxDeviceCount), "裝置上限必須大於零。");

            _maxDeviceCount = maxDeviceCount;
            _clock = clock ?? throw new ArgumentNullException(nameof(clock));
        }

        /// <summary>
        /// 根據方案月數計算從今日起算的到期日。
        /// </summary>
        /// <param name="planMonths">訂閱月數，必須為正整數。</param>
        /// <returns>到期日的 <see cref="DateTime"/>，時間部分固定為當天 23:59:59 UTC。</returns>
        /// <exception cref="ArgumentOutOfRangeException">當 <paramref name="planMonths"/> 小於等於零時擲出。</exception>
        public DateTime CalculateExpiryDate(int planMonths)
        {
            if (planMonths <= 0)
                throw new ArgumentOutOfRangeException(nameof(planMonths));

            // 使用 AddMonths 而非加天數，以正確處理不同月份天數差異（例如 2 月）
            var baseDate = _clock.UtcNow.Date.AddMonths(planMonths);

            // 到期日設為當天最後一刻，確保使用者可使用至該日結束
            return new DateTime(baseDate.Year, baseDate.Month, baseDate.Day, 23, 59, 59, DateTimeKind.Utc);
        }

        /// <summary>
        /// 嘗試將裝置加入訂閱，若已達裝置上限則拒絕並回傳 false。
        /// </summary>
        /// <param name="subscription">要操作的訂閱實體。</param>
        /// <param name="deviceId">要新增的裝置識別碼，不得為空白。</param>
        /// <returns>成功新增裝置時回傳 <c>true</c>；已達上限時回傳 <c>false</c>。</returns>
        /// <exception cref="ArgumentNullException">當 <paramref name="subscription"/> 為 null 時擲出。</exception>
        /// <exception cref="ArgumentException">當 <paramref name="deviceId"/> 為空白時擲出。</exception>
        public bool TryAddDevice(Subscription subscription, string deviceId)
        {
            if (subscription == null) throw new ArgumentNullException(nameof(subscription));
            if (string.IsNullOrWhiteSpace(deviceId)) throw new ArgumentException("裝置 ID 不得空白。", nameof(deviceId));

            if (subscription.DeviceIds.Count >= _maxDeviceCount)
                return false;

            subscription.DeviceIds.Add(deviceId);
            return true;
        }
    }

    /// <summary>
    /// 代表單一使用者的訂閱資料，包含方案內容與已綁定的裝置清單。
    /// </summary>
    public class Subscription
    {
        /// <summary>訂閱的唯一識別碼，由資料庫生成，不可手動指定。</summary>
        public int Id { get; set; }

        /// <summary>此訂閱歸屬的使用者識別碼。</summary>
        public string UserId { get; set; }

        /// <summary>訂閱到期的 UTC 時間；到期後系統應拒絕所有服務呼叫。</summary>
        public DateTime ExpiresAt { get; set; }

        /// <summary>
        /// 此訂閱目前已綁定的裝置識別碼清單。
        /// 清單大小受 <see cref="SubscriptionService"/> 的 maxDeviceCount 限制。
        /// </summary>
        public List<string> DeviceIds { get; set; } = new List<string>();
    }

    /// <summary>提供系統目前時間的抽象，允許在測試中替換為固定時間源。</summary>
    public interface ISystemClock
    {
        /// <summary>取得目前的 UTC 時間。</summary>
        DateTime UtcNow { get; }
    }
}
